reddit_api_async/connection.ml

```diff
@@ -3,16 +3,51 @@ open! Async
 open Reddit_api_kernel
 
 module Credentials = struct
+  module Password = struct
+    type t =
+      { client_id : string
+      ; client_secret : string
+      ; password : string
+      ; username : string
+      }
+    [@@deriving sexp]
+  end
+
+  module Userless_confidential = struct
+    type t =
+      { client_id : string
+      ; client_secret : string
+      }
+    [@@deriving sexp]
+  end
+
+  module Userless_public = struct
+    type t =
+      { client_id : string
+      ; device_id : string option [@sexp.option]
+      }
+    [@@deriving sexp]
+
+    let device_id_or_default device_id =
+      Option.value device_id ~default:"DO_NOT_TRACK_THIS_DEVICE"
+    ;;
```

Comment:
Is this function OK? Or should I use constant instead?

Reply:
Looks fine to me. I might suggest changing it to take a `Userless_public.t` instead of a `string option`, though it would make it a little more awkward to use in contexts where you're currently destructuring a `Userless_public.t` into its fields.

Reply:
Updated. Now the function takes `Userless_public.t`.